### PR TITLE
chore: Update dependencies

### PR DESCRIPTION
**Pull request by jordan-ms**
This commit updates the dependencies for flok-core. It also updates the project to use the latest version of yarn. The only code changes, which are minor, update types within flok-core to be TypeScript interfaces.

I also pinned flak-core's `prettier` version to an exact micro version, since this is the recommended practice according to the prettier docs. 

Since there is no test suite present, I tested locally by creating a production build and running the server. Everything works :)

### CODE REVIEW

**Comment by jordan-ms** [comment]
@munshkr Came across this project while watching a video on hydra.js. Thought I'd lend a hand. Could you please review this PR when you have time? I'd be happy to help with upgrading dependencies for the other packages too.

**Review by munshkr — CHANGES_REQUESTED** [review]
Hey! Thank you so much for contributing, I really appreaciate it. The only thing I'd change is the package version change, otherwise everything looks good!

File: packages/core/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "flok-core",
   "description": "Core package of flok, contains Pub/Sub functionality and CLI-based client",
-  "version": "0.4.10",
+  "version": "1.0.0",
```
[the remark this answers is not in the document]
Comment (jordan-ms) [reply packages/core/package.json:4]: Sounds good to me! I can make those changes right now.

[the remark this answers is not in the document]
Comment (jordan-ms) [reply packages/core/package.json]: Alrighty. Bumped the patch version in this commit: 7dbe9a5

**Review by munshkr — APPROVED** [review]

**Comment by munshkr** [comment]
Thanks!

**Pull request merged**